pkg/kopia/command/storage/secret_utils.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2022 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+type LocType string
+
+const (
+	// Location secret keys
+	bucketKey        = "bucket"
+	endpointKey      = "endpoint"
+	prefixKey        = "prefix"
+	regionKey        = "region"
+	skipSSLVerifyKey = "skipSSLVerify"
+	typeKey          = "type"
+
+	// Location types
+	LocTypeS3        LocType = "s3"
+	LocTypeGCS       LocType = "gcs"
+	LocTypeAzure     LocType = "azure"
+	LocTypeFilestore LocType = "filestore"
+)
+
+func getBucketNameFromMap(m map[string][]byte) string {
+	return string(m[bucketKey])
+}
+
+func getEndpointFromMap(m map[string][]byte) string {
+	return string(m[endpointKey])
+}
+
+func getPrefixFromMap(m map[string][]byte) string {
+	return string(m[prefixKey])
+}
+
+func getRegionFromMap(m map[string][]byte) string {
+	return string(m[regionKey])
+}
+
+func checkSkipSSLVerifyFromMap(m map[string][]byte) bool {
+	v := string(m[skipSSLVerifyKey])
+	return v == "true"
+}
+
+func locationType(m map[string][]byte) LocType {
+	return LocType(m[typeKey])
+}
+
+// GetMapForLocationValues return a map with valid keys
+// for different location values
+func GetMapForLocationValues(
```

Comment:
What did we decide about the validation of the location secret? Is that something we'll add to the repo server controller?
cc: @kale-amruta

Reply:
Should we keep them separate from these wrappers, at some higher level? We can avoid pod creation as well if validation fails.

Reply:
Yes, we should have some kinda validation definitely. Wanted to make sure we are aware of it 👍🏼